src/services/artifact/artifact.ts

```diff
@@ -1,10 +1,25 @@
 import * as artifact from '@actions/artifact';
+import { exec } from '@actions/exec';
 import * as fs from 'fs';
 import * as glob from '@actions/glob';
 
 class Artifact {
   async uploadLogs() {
     const basePath = '/home/runner/snap/charmcraft/common/cache/charmcraft/log';
+    const sudoPath = '/root/snap/charmcraft/common/cache/charmcraft/log';
+    // We're running some charmcraft commands as sudo as others as a
+    // regular user - we want to capture both.
+
+    // First check if the path created by sudo invocations of charmcraft
+    // exists.
+    const dirExistsExitCode = await exec('sudo', ['test', '-d', sudoPath]);
```

Comment:
Need to add `{ignoreReturnCode : true}` options or non-zero return will cause an error. https://github.com/actions/toolkit/blob/d9347d4ab99fd507c0b9104b2cf79fb44fcc827d/packages/exec/src/interfaces.ts#L28

Reply:
Thx, will update